src/components/Combobox/EntityCombobox.js

```diff
@@ -0,0 +1,144 @@
+    renderItemsAppended: PropTypes.func,
+    renderItemsPrepended: PropTypes.func,
+    renderListbox: PropTypes.func,
+    search: PropTypes.string,
+    items: PropTypes.arrayOf(itemTypeEntity),
+    onSearch: PropTypes.func.isRequired,
+    onSelect: PropTypes.func.isRequired,
+    selectedItems: PropTypes.arrayOf(itemTypeEntity),
+  }
+
+  static defaultProps = {
+    closeOnSelect: true,
+    comboboxClassName: null,
+    height: 5,
+    isLoading: false,
+    isMultiSelect: false,
+    items: [],
+    labelListbox: undefined,
+    renderInput: (inputProps, opts) => {
+      const { isMultiSelect, makeSelectHandler, selectedItems } = opts;
+      const { onKeyDown, value, ...rest } = inputProps;
+
+      const len = selectedItems.length;
+      const hasSingleSelection = !isMultiSelect && len === 1;
+
+      let inputValue = value;
+      let activeId = null;
+      let selectedItem;
+
+      if (hasSingleSelection) {
+        [selectedItem] = selectedItems;
+        const { id, label } = selectedItem;
+        inputValue = label;
+        activeId = id;
+      }
+
+      return (
+        <InputRaw
+          {...rest}
+          aria-activedescendant={!hasSingleSelection && activeId ? activeId : null}
+          aria-autocomplete="list"
+          className="slds-input slds-combobox__input slds-combobox__input-value"
+          iconLeft={selectedItem && (
+            <Icon
+              className="slds-combobox__input-entity-icon"
+              icon={selectedItem.icon.icon}
+              sprite={selectedItem.icon.sprite}
+            />
+          )}
+          iconRight={hasSingleSelection ? 'clear' : 'search'}
+          iconRightOnClick={hasSingleSelection ? makeSelectHandler(activeId) : null}
+          onKeyDown={makeInputAddHandler(onKeyDown, opts)}
+          readOnly={hasSingleSelection}
+          value={inputValue}
+        />
+      );
+    },
+    renderItem: ({ id, ...resultProps }, opts) => {
+      const { makeSelectHandler, search, selectedItems } = opts;
+
+      return (
+        <EntityDropdownItem
+          {...resultProps}
+          highlight={search}
+          isMultiSelect={!isEmpty(selectedItems)}
+          key={id}
+          id={id}
+          onSelect={makeSelectHandler(id)}
+        />
+      );
+    },
+    renderItemsAppended: null,
+    renderItemsPrepended: null,
+    renderListbox: listboxProps => <ComboboxListbox {...listboxProps} />,
+    search: '',
+    selectedItems: [],
+  }
+
+  onSearch = (input) => {
+    const { onSearch } = this.props;
+    const val = isObject(input) ? input.target.value : input;
+    onSearch(val, { isClear: false });
+  }
+
+  onSelect = (id, opts) => {
+    const { onSearch, onSelect } = this.props;
+    onSelect(id, opts);
+    onSearch('', { isClear: true });
+  }
+
+  renderSearchIndicator = () => {
+    const { renderItemsPrepended, search } = this.props;
+    if (search.trim().length < 2) return null;
+    return (
+      <React.Fragment>
```

Comment:
what are gains of using `<React.Fragment>` over `<>`?

Reply:
We need to update the babel tooling in rlds before we can do that. I'll open a separate PR for that